server/api/store.go

```diff
@@ -610,6 +610,47 @@ func (h *storesHandler) GetStoreLimitScene(w http.ResponseWriter, r *http.Reques
 	h.rd.JSON(w, http.StatusOK, scene)
 }
 
+// StoreProgress contains status about a progress.
+type StoreProgress struct {
+	StoreID      uint64  `json:"store_id"`
+	Action       string  `json:"action"`
+	Progress     float64 `json:"progress"`
+	CurrentSpeed float64 `json:"current_speed"`
+	LeftSeconds  float64 `json:"left_seconds"`
+}
+
+// @Tags stores
+// @Summary Get store progress in the cluster.
+// @Produce json
+// @Success 200 {object} StoreProgress
+// @Failure 400 {string} string "The input is invalid."
+// @Failure 500 {string} string "PD server failed to proceed the request."
+// @Router /stores/{id}/progress [get]
+func (h *storesHandler) GetStoreProgressByID(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	storeID, errParse := apiutil.ParseUint64VarsField(vars, "id")
+	if errParse != nil {
+		apiutil.ErrorResp(h.rd, w, errcode.NewInvalidInputErr(errParse))
+		return
+	}
+
+	action := r.URL.Query().Get("action")
+	if err := checkProgressAction(action); err != nil {
+		h.rd.JSON(w, http.StatusBadRequest, err.Error())
+		return
+	}
+	progress, currentSpeed, leftSeconds := h.Handler.GetStoreProgressByID(action, storeID)
```

Comment:
Same as below, `GetStoreProgressByID` need action, how get progress by `GetStoreProgressByAction` if I want to remove or add multiple nodes.

Reply:
The problem is that do we want to know the progress of each store? We can add another interface to return specified action progress.

Reply:
I think it's needed, because there may exist removing stores and preparing stores during the same time period, the user may separately check it.